### DISCUSSION TITLE
Tailwind Elements is stable

### DISCUSSION THREAD

**Discussion by filipkappa** [mdbootstrap/TW-Elements#1961] [Announcements]
We released the **[first 🥇 Stable Version](https://tailwind-elements.com/docs/standard/getting-started/changelog/)** of Tailwind Elements.

**[🕊 RETWEET OUR ANNOUNCEMENT & SPREAD THE WORD](https://twitter.com/TailwindElement/status/1704789131430371470
)**
**[▶ WATCH TE STABLE "FROM ZERO TO PRODUCTION" TUTORIAL](https://www.youtube.com/watch?v=6a69dFvg5bI&ab_channel=Keepcoding
)**

This is a perfect occassion to summarize what we managed to achieve with TE over the last couple of months.

---

### 🔌 Tailwind Elements easily integrates with:

- Angular
- Vue
- Express
- Django
- Laravel
- Next
- Nuxt
- Remix
- Solid
- Svelte

\+ [the community voted](https://github.com/mdbootstrap/Tailwind-Elements/discussions/1443) to create a dedicated ⚛ [React kit](https://tailwind-elements.com/docs/react/), which is beeing developed by our awesome contributors 💪

---

### ⚡ Optimization & fast-loading is simple

We took a **modular** approach to optimization in TE.

TailwindCSS is already great with CSS optimizaion in itself. We made sure that with TE, **[you are able to import only the crucial components & plugins you need](https://tailwind-elements.com/docs/standard/getting-started/optimization/#modules)**, so that your JS isn't bloated. Tailwind Elements will ensure that **your projects load fast & score well in SEO**.

Here is an interesting table covering the **representative examples** of some components imports.
The maximum possible optimization for the initial **weight of the JS** in the package, in an extreme case, is approximately **97.62%** reduction based on the absolute file size or 63.60% reduction based on the gzipped file size.

|  #  | Format | Components used in a Vite project |                                        Documentation link                                         | Init type | JS files weight [kB] | gzip weight [kB] | % of UMD (full init) |
| :-: | :----: | :-------------------------------: | :-----------------------------------------------------------------------------------------------: | :-------: | :------------------: | :--------------: | :------------------: |
|  1  |  UMD   |           Full UMD init           |                                           Full UMD init                                           |     -     |        675,21        |      190,83      |       100,00%        |
|  2  |   ES   |               Input               |            [see docs](https://tailwind-elements,com/docs/standard/forms/inputs/#basic)            |  initTE   |        33,74         |      10,42       |        5,00%         |
|  3  |   ES   |               Input               |            [see docs](https://tailwind-elements,com/docs/standard/forms/inputs/#basic)            |  via JS   |        24,47         |       8,17       |        3,62%         |
|  4  |   ES   |           Input, Ripple           |          [see docs](https://tailwind-elements,com/docs/standard/forms/login-form/#basic)          |  initTE   |        39,96         |      12,38       |        5,92%         |
|  5  |   ES   |           Input, Ripple           |          [see docs](https://tailwind-elements,com/docs/standard/forms/login-form/#basic)          |  via JS   |        30,75         |      10,15       |        4,55%         |
|  6  |   ES   |              Select               |               [see docs](https://tailwind-elements,com/docs/standard/forms/select/)               |  initTE   |        99,33         |      12,36       |        14,71%        |
|  7  |   ES   |              Select               |               [see docs](https://tailwind-elements,com/docs/standard/forms/select/)               |  via JS   |        90,04         |      26,39       |        13,34%        |
|  8  |   ES   |     Clipboard, Input, Ripple      |            [see docs](https://tailwind-elements,com/docs/standard/methods/clipboard/)             |  initTE   |        41,62         |       12,8       |        6,16%         |
|  9  |   ES   |     Clipboard, Input, Ripple      |            [see docs](https://tailwind-elements,com/docs/standard/methods/clipboard/)             |  via JS   |        32,48         |      10,57       |        4,81%         |
| 10  |   ES   |             Clipboard             |            [see docs](https://tailwind-elements,com/docs/standard/methods/clipboard/)             |  initTE   |        25,59         |       8,68       |        3,79%         |
| 11  |   ES   |             Clipboard             |            [see docs](https://tailwind-elements,com/docs/standard/methods/clipboard/)             |  via JS   |        16,09         |       6,32       |        2,38%         |
| 12  |   ES   |         LoadingManagement         | [see docs](https://tailwind-elements,com/docs/standard/methods/loading-management/#basic-example) |  initTE   |        27,71         |       9,17       |        4,10%         |
| 13  |   ES   |         LoadingManagement         | [see docs](https://tailwind-elements,com/docs/standard/methods/loading-management/#basic-example) |  via JS   |        18,29         |       6,86       |        2,71%         |

---

### 🎨 Theming & customization is powerful

You can **[easily define your own](https://tailwind-elements.com/docs/standard/getting-started/theming/)**:
- color palette
- type scale
- fonts
- breakpoints
- border radius 
- & more!

Also, **[dark mode](https://tailwind-elements.com/docs/standard/getting-started/dark-mode/)** 🕶 in TE works by default. TE detects the preferred theme & adjusts the 🎨 color pallete.
You can easily add a manual toggler for the users to switch the theme 🌞 -> 🌙

All elements are dark-mode friendly, and dark versions are also included in our **[design system Figma files](https://www.figma.com/community/file/1255241951305406231/material-minimal-design-system)**.

---

### 🌎 Internationalization (language switcher) is quick

If you need an easy way to develop a **multi-lingual website** TE is there for you.

Check out the **[live example](https://te-standard-i18n.mdbgo.io/#)** of a language switcher in a TE project.

With TE you are able to easily **[add i18n to your project](https://tailwind-elements.com/docs/standard/getting-started/internationalization/)** & start using directional classes & JSON language variants for your content.

---

### Thanks to contributors!

All of this was achieved thanks to our awesome 💪 contributors. Special thanks to:
- @juujisai 
- @Trochonovitz 
- @iprzybysz 
- @Martussky 
- @smolenski-mikolaj

**Comment by Durgeswarank** [comment]
how to i use tw elements for angular 17 , i get installed but idont knoew the working flow

**Reply by juujisai** [reply]
Hi! Checkout the Angular integration guide in our docs: https://tw-elements.com/docs/standard/integrations/angular-integration/

**Comment by Durgeswarank** [comment]
Thanks for your support, but I already done this properly. I got same error
again. My problem is for older angular version it works fine.
Currently Im using version 17 of angular. And I was updated my cli and
angular versions respectively.  And I tried both default standalone
structure of angular 17 project and no standalone type . But both are have
same issues.
But here I want to say  one thing in tw elements have changed its install
steps . We can see the tailwind. config.js  code snippet was changed .
And in earlier i was insert a link in my index.html page . But now that
link is not available in tw elements documentation.
   Please check it and help me !

**Comment by iprzybysz** [comment]
Hi @Durgeswarank, please create a separate issue or discussion and there clearly describe your problem. Attach also image or content of your error message. Then, I'll try to help you.

**Comment by Durgeswarank** [comment]
Hi I cleared that .
I disable SSR during project generation. Now this error was gone. 🙂
